google_guest_agent/network/manager/common_linux.go

```diff
@@ -0,0 +1,72 @@
+//  Copyright 2024 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package manager is responsible for detecting the current network manager service, and
+// writing and rolling back appropriate configurations for each network manager service.
+package manager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/run"
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+)
+
+// configureIPv6 sets the appropriate system variable for IPv6 configuration for all provided
+// interfaces. The interfaces provided should only consist of IPv6-supporting interfaces.
+// The primaryInterface parameter is used to ensure tentative IPs are resolved before
+// IPv6 leases are released or obtained.
+func configureIPv6(ctx context.Context, ipv6Interfaces []string, primaryInterface metadata.NetworkInterfaces) error {
+	if primaryInterface.DHCPv6Refresh != "" {
+		// Wait for tentative IPs as part of SLAAC to resolve for primary interface.
+		primaryNic, err := GetInterfaceByMAC(primaryInterface.Mac)
+		if err != nil {
+			return fmt.Errorf("error getting primary interface: %v", err)
+		}
+
+		tentative := []string{"-6", "-o", "a", "s", "dev", primaryNic.Name, "scope", "link", "tentative"}
+		for i := 0; i < 5; i++ {
+			res := run.WithOutput(ctx, "ip", tentative...)
+			if res.ExitCode == 0 && res.StdOut == "" {
+				break
+			}
+			time.Sleep(1 * time.Second)
+		}
+	}
+
+	for _, iface := range ipv6Interfaces {
+		// Set the appropriate system values.
+		val := fmt.Sprintf("net.ipv6.conf.%s.accept_ra_rt_info_max_plen=128", iface)
+		if err := run.Quiet(ctx, "sysctl", val); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// interfaceNames extracts the names of the network interfaces from the provided list
+// of network interfaces.
+func interfaceNames(nics []metadata.NetworkInterfaces) ([]string, error) {
```

Comment:
since implementation is not specific to Linux should it be in `common.go` rather than `common_linux.go` ?